File: frontend/src/components/OnboardingScreen.js
import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { useLanguage } from '../App';
import { Bitcoin, Shield, CreditCard } from 'lucide-react';
import NubixLogo from './NubixLogo';

const OnboardingScreen = () => {
  const navigate = useNavigate();
  const { completeOnboarding } = useLanguage();
  const [currentPage, setCurrentPage] = useState(0);

  const pages = [
    {
      icon: Bitcoin,
      title: 'Welcome to NubiX',
      description: 'Your trusted platform for buying cryptocurrency with Sudanese Pounds through secure transactions.',
      color: 'text-primary-500'
    },
    {
      icon: Shield,
      title: 'Secure Trading',
      description: 'Advanced security features with KYC verification and multi-factor authentication for your protection.',
      color: 'text-success'
    },
    {
      icon: CreditCard,
      title: 'Bank Integration',
      description: 'Seamless integration with Bank of Khartoum through Bankak payment system for easy transactions.',
      color: 'text-secondary-600'
    }
  ];

  const handleNext = () => {
    if (currentPage < pages.length - 1) {
      setCurrentPage(currentPage + 1);
    } else {
      handleComplete();
    }
  };

  const handlePrevious = () => {
    if (currentPage > 0) {
      setCurrentPage(currentPage - 1);
    }
  };

  const handleComplete = () => {
    completeOnboarding();
    navigate('/login');
  };

  return (
    <div className="min-h-screen bg-gradient-to-br from-gray-50 to-gray-100">
      {/* Header with better spacing */}
      <div className="flex justify-between items-center p-8">
        <NubixLogo size="md" showText={true} />
        
        <button
          onClick={handleComplete}
          className="text-primary-600 font-medium hover:text-primary-700 nubix-transition bg-white px-6 py-2 rounded-full shadow-nubix"
        >
          Skip
        </button>
      </div>

      {/* Content with improved spacing */}
      <div className="flex-1 flex flex-col justify-center px-8 py-16">
        <div className="max-w-lg mx-auto text-center">
          {/* Icon with better spacing */}
          <div className="mb-12">
            <div className="w-28 h-28 bg-white rounded-3xl shadow-nubix-lg flex items-center justify-center mx-auto mb-8">
              {React.createElement(pages[currentPage].icon, {
                size: 56,
                className: pages[currentPage].color
              })}
            </div>
          </div>

          {/* Title with improved typography */}
          <h1 className="text-4xl font-display font-bold text-gray-900 mb-6 leading-tight">
            {pages[currentPage].title}
          </h1>

          {/* Description with better spacing */}
          <p className="text-gray-600 text-xl leading-relaxed px-4">
            {pages[currentPage].description}
          </p>
        </div>
      </div>

      {/* Bottom Navigation with improved spacing */}
      <div className="p-8">
        {/* Page Indicators */}
        <div className="flex justify-center mb-8">
          {pages.map((_, index) => (
            <div
              key={index}
              className={`w-3 h-3 rounded-full mx-2 nubix-transition ${
                currentPage === index 
                  ? 'bg-primary-500 w-10' 
                  : 'bg-gray-300'
              }`}
            />
          ))}
        </div>

        {/* Navigation Buttons with better spacing */}
        <div className="max-w-md mx-auto">
          <div className="flex space-x-4">
            {currentPage > 0 && (
              <button
                onClick={handlePrevious}
                className="flex-1 nubix-btn-secondary rounded-2xl py-4 text-lg"
              >
                Previous
              </button>
            )}
            <button
              onClick={handleNext}
              className="flex-1 nubix-btn-primary rounded-2xl py-4 text-lg"
            >
              {currentPage === pages.length - 1 ? 'Get Started' : 'Next'}
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default OnboardingScreen;

File: lib/features/onboarding/onboarding_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import 'package:go_router/go_router.dart';
import '../../core/providers/auth_provider.dart';
import '../../core/providers/language_provider.dart';
import '../../core/theme/app_colors.dart';
import '../../core/widgets/nubix_logo.dart';

class OnboardingScreen extends StatefulWidget {
  const OnboardingScreen({Key? key}) : super(key: key);

  @override
  State<OnboardingScreen> createState() => _OnboardingScreenState();
}

class _OnboardingScreenState extends State<OnboardingScreen> {
  final PageController _pageController = PageController();
  int _currentPage = 0;

  final List<OnboardingPage> _pages = [
    OnboardingPage(
      title: 'Welcome to NUBIX',
      titleAr: 'مرحباً بك في نوبكس',
      description: 'Your trusted platform for buying cryptocurrency with Sudanese Pounds',
      descriptionAr: 'منصتك الموثوقة لشراء العملات المشفرة بالجنيه السوداني',
      icon: Icons.currency_bitcoin,
      color: AppColors.bitcoin,
    ),
    OnboardingPage(
      title: 'Secure Trading',
      titleAr: 'تداول آمن',
      description: 'Advanced security features with KYC verification and biometric authentication',
      descriptionAr: 'ميزات أمان متقدمة مع التحقق من الهوية والمصادقة البيومترية',
      icon: Icons.security,
      color: AppColors.success,
    ),
    OnboardingPage(
      title: 'Bank Integration',
      titleAr: 'تكامل مصرفي',
      description: 'Seamless integration with Bank of Khartoum through Bankak payment system',
      descriptionAr: 'تكامل سلس مع بنك الخرطوم من خلال نظام بنكك للدفع',
      icon: Icons.account_balance,
      color: AppColors.primary,
    ),
    OnboardingPage(
      title: 'Choose Your Language',
      titleAr: 'اختر لغتك',
      description: 'Select your preferred language to get started',
      descriptionAr: 'اختر لغتك المفضلة للبدء',
      icon: Icons.language,
      color: AppColors.secondary,
    ),
  ];

  @override
  void dispose() {
    _pageController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: Column(
          children: [
            // Header with logo and skip button
            Padding(
              padding: const EdgeInsets.all(20),
              child: Row(
                mainAxisAlignment: MainAxisAlignment.spaceBetween,
                children: [
                  const NubixLogo(size: 40, showText: true),
                  if (_currentPage < _pages.length - 1)
                    TextButton(
                      onPressed: () => _goToLanguageSelection(),
                      child: const Text('Skip'),
                    ),
                ],
              ),
            ),
            
            // PageView
            Expanded(
              child: PageView.builder(
                controller: _pageController,
                onPageChanged: (index) {
                  setState(() {
                    _currentPage = index;
                  });
                },
                itemCount: _pages.length,
                itemBuilder: (context, index) {
                  if (index == _pages.length - 1) {
                    return _buildLanguageSelectionPage();
                  }
                  return _buildOnboardingPage(_pages[index]);
                },
              ),
            ),
            
            // Bottom navigation
            Padding(
              padding: const EdgeInsets.all(20),
              child: Column(
                children: [
                  // Page indicators
                  Row(
                    mainAxisAlignment: MainAxisAlignment.center,
                    children: List.generate(
                      _pages.length,
                      (index) => AnimatedContainer(
                        duration: const Duration(milliseconds: 300),
                        margin: const EdgeInsets.symmetric(horizontal: 4),
                        width: _currentPage == index ? 24 : 8,
                        height: 8,
                        decoration: BoxDecoration(
                          color: _currentPage == index 
                              ? AppColors.primary 
                              : AppColors.outline,
                          borderRadius: BorderRadius.circular(4),
                        ),
                      ),
                    ),
                  ),
                  
                  const SizedBox(height: 24),
                  
                  // Navigation buttons
                  if (_currentPage < _pages.length - 1) ...[
                    SizedBox(
                      width: double.infinity,
                      child: ElevatedButton(
                        onPressed: _nextPage,
                        child: Text(_currentPage == _pages.length - 2 ? 'Continue' : 'Next'),
                      ),
                    ),
                    if (_currentPage > 0) ...[
                      const SizedBox(height: 12),
                      SizedBox(
                        width: double.infinity,
                        child: OutlinedButton(
                          onPressed: _previousPage,
                          child: const Text('Previous'),
                        ),
                      ),
                    ],
                  ],
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildOnboardingPage(OnboardingPage page) {
    final languageProvider = Provider.of<LanguageProvider>(context);
    final isArabic = languageProvider.isArabic;
    
    return Padding(
      padding: const EdgeInsets.all(20),
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          // Icon
          Container(
            width: 120,
            height: 120,
            decoration: BoxDecoration(
              color: page.color.withOpacity(0.1),
              shape: BoxShape.circle,
            ),
            child: Icon(
              page.icon,
              size: 60,
              color: page.color,
            ),
          ),
          
          const SizedBox(height: 40),
          
          // Title
          Text(
            isArabic ? page.titleAr : page.title,
            style: Theme.of(context).textTheme.displaySmall?.copyWith(
              fontWeight: FontWeight.bold,
              color: AppColors.onSurface,
            ),
            textAlign: TextAlign.center,
          ),
          
          const SizedBox(height: 16),
          
          // Description
          Text(
            isArabic ? page.descriptionAr : page.description,
            style: Theme.of(context).textTheme.bodyLarge?.copyWith(
              color: AppColors.onSurfaceVariant,
              height: 1.5,
            ),
            textAlign: TextAlign.center,
          ),
        ],
      ),
    );
  }

  Widget _buildLanguageSelectionPage() {
    return Padding(
      padding: const EdgeInsets.all(20),
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          const Icon(
            Icons.language,
            size: 80,
            color: AppColors.secondary,
          ),
          
          const SizedBox(height: 40),
          
          Text(
            'Choose Your Language',
            style: Theme.of(context).textTheme.displaySmall?.copyWith(
              fontWeight: FontWeight.bold,
            ),
            textAlign: TextAlign.center,
          ),
          
          const SizedBox(height: 8),
          
          Text(
            'اختر لغتك المفضلة',
            style: Theme.of(context).textTheme.displaySmall?.copyWith(
              fontWeight: FontWeight.bold,
            ),
            textAlign: TextAlign.center,
          ),
          
          const SizedBox(height: 40),
          
          // Language selection buttons
          Column(
            children: [
              SizedBox(
                width: double.infinity,
                child: ElevatedButton.icon(
                  onPressed: () => _selectLanguageAndContinue('en'),
                  icon: const Text('🇺🇸', style: TextStyle(fontSize: 24)),
                  label: const Text('English'),
                  style: ElevatedButton.styleFrom(
                    padding: const EdgeInsets.all(16),
                  ),
                ),
              ),
              
              const SizedBox(height: 16),
              
              SizedBox(
                width: double.infinity,
                child: ElevatedButton.icon(
                  onPressed: () => _selectLanguageAndContinue('ar'),
                  icon: const Text('🇸🇩', style: TextStyle(fontSize: 24)),
                  label: const Text('العربية'),
                  style: ElevatedButton.styleFrom(
                    padding: const EdgeInsets.all(16),
                  ),
                ),
              ),
            ],
          ),
        ],
      ),
    );
  }

  void _nextPage() {
    if (_currentPage < _pages.length - 1) {
      _pageController.nextPage(
        duration: const Duration(milliseconds: 300),
        curve: Curves.easeInOut,
      );
    }
  }

  void _previousPage() {
    if (_currentPage > 0) {
      _pageController.previousPage(
        duration: const Duration(milliseconds: 300),
        curve: Curves.easeInOut,
      );
    }
  }

  void _goToLanguageSelection() {
    _pageController.animateToPage(
      _pages.length - 1,
      duration: const Duration(milliseconds: 300),
      curve: Curves.easeInOut,
    );
  }

  void _selectLanguageAndContinue(String languageCode) {
    final languageProvider = Provider.of<LanguageProvider>(context, listen: false);
    final authProvider = Provider.of<AuthProvider>(context, listen: false);
    
    // Set selected language
    if (languageCode == 'ar') {
      languageProvider.setArabic();
    } else {
      languageProvider.setEnglish();
    }
    
    // Mark onboarding as complete
    authProvider.setFirstTimeComplete();
    
    // Navigate to login
    context.go('/login');
  }
}

class OnboardingPage {
  final String title;
  final String titleAr;
  final String description;
  final String descriptionAr;
  final IconData icon;
  final Color color;

  const OnboardingPage({
    required this.title,
    required this.titleAr,
    required this.description,
    required this.descriptionAr,
    required this.icon,
    required this.color,
  });
}